### PR TITLE
SBERDOMA-394 ticket status color's backend forwarding

### PR DESCRIPTION
**Pull request by Dimitreee**

### CODE REVIEW

File: apps/condo/domains/ticket/schema/TicketStatus.js [hunk 1]
```diff
@@ -30,6 +48,38 @@ const TicketStatus = new GQLListSchema('TicketStatus', {
...
         },
+
+        colors: {
+            schemaDoc: 'Status colors, includes primary (font color), secondary (background color), additional (border color), all colors presented in HEX',
+            type: Virtual,
+            graphQLReturnType: 'JSON',
+            resolver: async (item) => {
+                if (!get(item, 'type')) {
+                    const error = 'Error while trying to find ticket status color'
+                    console.error(error)
```
Comment (AntonAL) [inline apps/condo/domains/ticket/schema/TicketStatus.js:59]: remove `console.log`, please

File: apps/condo/domains/ticket/schema/TicketStatus.js [hunk 2]
```diff
@@ -30,6 +48,38 @@ const TicketStatus = new GQLListSchema('TicketStatus', {
...
+
+                    throw new Error(error)
+                }
+
+                const statusColors = STATUS_SELECT_COLORS[item.type]
+
+                if (!statusColors) {
+                    const error = 'Error while trying to find ticket status color: no ticketStatus colors found'
+                    console.error(error)
```
Comment (AntonAL) [inline apps/condo/domains/ticket/schema/TicketStatus.js:68]: remove `console.log`, please

File: apps/condo/domains/ticket/schema/TicketStatus.js [hunk 3]
```diff
@@ -30,6 +48,38 @@ const TicketStatus = new GQLListSchema('TicketStatus', {
...
+
+                    throw new Error(error)
+                }
+
+                if (!ticketStatusColorsJsonValidator(statusColors)){
+                    const error = `${JSON_SCHEMA_VALIDATION_ERROR}] invalid json structure`
+                    console.error(error)
```
Comment (AntonAL) [inline apps/condo/domains/ticket/schema/TicketStatus.js:75]: remove `console.error`, please

File: apps/condo/domains/ticket/schema/TicketStatus.js [hunk 4]
```diff
@@ -30,6 +48,38 @@ const TicketStatus = new GQLListSchema('TicketStatus', {
             type: Text,
             isRequired: true,
         },
+
+        colors: {
+            schemaDoc: 'Status colors, includes primary (font color), secondary (background color), additional (border color), all colors presented in HEX',
+            type: Virtual,
+            graphQLReturnType: 'JSON',
```
Comment (AntonAL) [inline apps/condo/domains/ticket/schema/TicketStatus.js:55]: Can you use a type `Json` from `@core/keystone/fields`?

**Review by AntonAL — COMMENTED** [review]
Please, write tests for `TicketStatus.colors` field.

**Comment by Dimitreee** [comment]
> Please, write tests for `TicketStatus.colors` field.

done

File: apps/condo/domains/ticket/schema/TicketStatus.js [hunk 5]
```diff
@@ -30,6 +48,29 @@ const TicketStatus = new GQLListSchema('TicketStatus', {
...
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/TicketStatus.js:55]: What about something like:

```
      extendGraphQLTypes: [`type TicketStatusColor { primary: String, secondary: String, additional: String }`],
      graphQLReturnType: `TicketStatusColor`,
      graphQLReturnFragment: `{
        primary
        secondary
        additional
      }`,

```

**Review by pahaz — APPROVED** [review]

**Review by dkoviazin — APPROVED** [review]
Ok

**Review by AntonAL — APPROVED** [review]

**Pull request merged**